### PR TITLE
Do not index array lengths of inner arrays of nested object arrays

### PR DESCRIPTION
**Pull request by jeeminso**
## Summary of the changes / Why this improves CrateDB

We do not have names for inner arrays of nested object arrays so we cannot index/query them(_directly_). Currently they are indexed as if they are the lengths of the top level array resulting in multple array lengths of the top level array. The fix is simple to prevent indexing array lengths of inner arrays.

Follows https://github.com/crate/crate/pull/16688

## Checklist

 - [x] Added an entry in the latest `docs/appendices/release-notes/<x.y.0>.rst` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in the latest `docs/appendices/release-notes/<x.y.0>.rst`
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by jeeminso** [comment]
Hi @romseygeek , could you please help double check if this breaks `ArrayOfObjectIndexer` in any way?

**Review by romseygeek — COMMENTED** [review]
Could you add a test demonstrating a query that would previously have returned the wrong results?

**Comment by jeeminso** [comment]
Thank you, I found another bug https://github.com/crate/crate/pull/16969 while trying to add more tests. I think this could wait for few days.

**Comment by jeeminso** [comment]
Without this fix, the following asserts fail:
```
assertThat(tester.runQuery("o", "array_length(o, 1) = 1")).isEmpty();
assertThat(tester.runQuery("o", "array_length(o, 1) = 3")).isEmpty();
assertThat(tester.runQuery("o", "array_length(o['a'], 1) = 1")).isEmpty();
assertThat(tester.runQuery("o", "array_length(o['a'], 1) = 3")).isEmpty();
```

**Comment by jeeminso** [comment]
Could you review again @romseygeek ? Added a test with queries that would have failed.

**Comment by BaurzhanSakhariev** [comment]
> Without this fix, the following asserts fail:

Shouldn't we add a change entry?

**Review by romseygeek — APPROVED** [review]
As @BaurzhanSakhariev says, we need a changes entry, but all LGTM other than that.  Thanks!

**Comment by jeeminso** [comment]
Sorry did not make it clear, it is a regression from https://github.com/crate/crate/pull/16569, so we do not need a changes entry.

File: server/src/test/java/io/crate/lucene/NestedArrayLuceneQueryBuilderTest.java [hunk 1]
```diff
@@ -132,4 +133,50 @@ public void test_any_not_equals_on_array_literal_and_nested_array_ref_with_autom
         var query = convert("[1] != any(c)");
         assertThat(query.toString()).isEqualTo("([1] <> ANY(array_unnest(c)))");
     }
+
+    @Test
+    public void test_lengths_of_nested_array_of_objects() throws Exception {
+        QueryTester.Builder builder = new QueryTester.Builder(
+            THREAD_POOL,
+            clusterService,
+            Version.CURRENT,
```
Comment (jeeminso) [inline server/src/test/java/io/crate/lucene/NestedArrayLuceneQueryBuilderTest.java:142]: Just double checked, updating this to `V_5_9_0` still passes.

Comment (jeeminso) [reply server/src/test/java/io/crate/lucene/NestedArrayLuceneQueryBuilderTest.java:142]: If I create the table on `5.9` and execute the queries on `5.10`, there are failures. I will create a follow up PR.

**Pull request merged**